userspace/falco/configuration.cpp

```diff
@@ -183,8 +183,38 @@ void falco_configuration::load_engine_config(const std::string& config_name, con
 	m_changes_in_engine_config = true;
 }
 
-void falco_configuration::load_yaml(const std::string& config_name, const yaml_helper& config)
+std::unordered_set<std::string> extractKeys(const YAML::Node& node) {
+    std::unordered_set<std::string> keys;
+
+    for (auto it = node.begin(); it != node.end(); ++it) {
+        const auto& key = it->first.as<std::string>();
+        keys.insert(key);
+    }
+
+    return keys;
+}
+
+void validateLoadedYAML(const YAML::Node& loadedYAML, const std::unordered_set<std::string>& fixedSchemaKeys)
 {
+    std::unordered_set<std::string> loadedYAMLKeys = extractKeys(loadedYAML);
+
+    // Check for keys in loaded YAML that are not in the fixed schema
+    for (const auto& key : loadedYAMLKeys) {
+        if (fixedSchemaKeys.find(key) == fixedSchemaKeys.end()) {
+            throw std::logic_error("Error: Key '" + key + "' does not exist in the fixed schema");
+        }
+    }
+}
+void falco_configuration::load_yaml(const std::string& config_name, const yaml_helper& config)
+{   
+    YAML::Node fixedSchema = YAML::LoadFile("userspace/falco/fixed.yaml");
```

Comment:
This can work only when in development mode. A better solution would be to embed the file into the Falco binary so as not to load an external file at runtime.

Reply:
hey @leogr,
Thanks for reviewing. I will embed file into Falco binary and along with that try to find other validation mechanisms too.